storm-client/src/jvm/org/apache/storm/executor/Executor.java

```diff
@@ -270,6 +273,15 @@ public ExecutorShutdown execute() throws Exception {
 
     @Override
     public void accept(Object event) {
+        if (this.needToRefreshCreds) {
```

Comment:
Another issue about checking the value here is it might be blocked.
This `accept` is only called when `receiveQueue.consume` is called. But it could be blocked so credential update will be delayed.

Reply:
moved this to separate method and being called fro withing `BoltExecutor` and `SpoutExecutor` call at the beginning of each iteration in the asyncloop. This would ensure that credentials are updated on executor irrespective of back pressure or no tuples to process scenarios. For backward compatibility and agreement about invoking _setCreentials_ on user implementation on _ICredentialsListener_ from executor thread, if Executor thread is stuck in _execute_ or _nextTuple_  methods delays can not be avoided. But otherwise we should be able to get credentials update on next iteration.